File: test_api.js
const https = require('https');

const data = JSON.stringify({
  text: 'The committee must comply immediately with the new directive.'
});

const options = {
  hostname: 'pmddproject.vercel.app',
  path: '/api/analyze',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const start = Date.now();
const req = https.request(options, (res) => {
  console.log('Status:', res.statusCode);
  console.log('Time:', Date.now() - start, 'ms');
  
  let body = '';
  res.on('data', (d) => {
    body += d;
  });
  
  res.on('end', () => {
    console.log('Body:', body.substring(0, 300));
  });
});

req.on('error', (error) => {
  console.error('Error:', error);
});

req.write(data);
req.end();

File: test_api.py
import urllib.request
import json

url = "https://pmddproject.vercel.app/api/analyze"
data = json.dumps({"text": "This is a simple test sentence for the backend."}).encode('utf-8')
req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})

try:
    with urllib.request.urlopen(req) as response:
        print("Status:", response.status)
        print("Response:", response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print("HTTP Error:", e.code)
    print("Error Response:", e.read().decode('utf-8'))
except Exception as e:
    print("Exception:", str(e))
